build/commands/lib/config.js

```diff
@@ -532,10 +532,14 @@ Config.prototype.buildArgs = function () {
     generate_about_credits: true,
   }
 
-  if (this.isOfficialBuild()) {
+  if (this.isOfficialBuild() || getEnvConfig(['enable_updater'])) {
```

Comment:
@goodov  what are your thoughts on this? If are gonna pass through this `gn` value through `.env`, we may as well do away with the the magic setting of it for `isOfficialBuild`, and move this into `.env` files in the CI. What do you think?

If you think that could be the case, then we would just do:

```
args.enable_updater = getEnvConfig(['enable_updater'])
```

Reply:
I agree with @bridiver that having defaults on CI is not great. I'm fine with the change in this PR if we look at this in the context of our existing approach.

What we should aim for in my opinion is to have `Release_mac.gni` in brave-core, which would have defaults listed in a clear way:
```
enable_updater = true
```
when done right, these values will be available for overriding with `.env` values in `args_generated.gni` like so:
```
import("//brave/build/configs/Release_mac.gni")

args.enable_updater = true/false # if set in .env
```

Reply:
I feel `.env` vars make sense for secrets, but little to no sense for boolean or public values.

Reply:
I think for this PR what we have here is fine and we should move the discussion about improvements somewhere else

Reply:
actually re `npm run build ... -- --gn enable_updater:true --gn use_prebuilt_omaha4:true` - how do we handle this stuff now in CI? If we're using `.env` then we should use this PR and keep things consistent for now, otherwise we could just pass the args directly. cc @wknapik

Reply:
I mean I'm fine with this PR, but not sure if it's actually necessary here or not